### PR TITLE
Fix/nested tables

### PR DESCRIPTION
**Pull request by AlexanderMann**
# Motivation

https://github.com/datamill-co/target-postgres/issues/7
https://github.com/datamill-co/target-postgres/issues/42
https://github.com/datamill-co/target-postgres/issues/58
https://github.com/datamill-co/target-postgres/issues/61

Nested table issues. Each known failing problem has been added as a commit.

## Notes

This is pseudo-branched off of https://github.com/datamill-co/target-postgres/pull/60 due to needing more complex `db_cleanup` logic.

As such there _is_ a failing test kicking around in here which is _not_ related to the problems this pr will _ultimately_ be addressing.

## Suggested Musical Pairing

https://soundcloud.com/phoenix/heatwave

### CODE REVIEW

File: target_postgres/json_schema.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from jsonschema import Draft4Validator
 from jsonschema.exceptions import SchemaError
 
+NULL = 'null'
```
Comment (awm33) [inline target_postgres/json_schema.py]: Should we create one for each JSON Schema type we use?

File: tests/test_sandbox.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    cursor.execute("SELECT column_name, data_type, is_nullable FROM information_schema.columns " + \
+                   "WHERE table_schema = 'public' and table_name = '{}';".format(
+                       table_name))
+    columns = cursor.fetchall()
+
+    assert (not columns and not expected_column_tuples) \
+           or set(columns) == expected_column_tuples
+
+
+class BigCommerceStream:
```
Comment (awm33) [inline tests/test_sandbox.py]: Where's the generator for this?

File: target_postgres/json_schema.py (hunk 1 above)
Comment (AlexanderMann) [reply target_postgres/json_schema.py]: Prolly. I was doing the minimal change first. I think there's reason to do this regardless though

File: tests/test_sandbox.py (hunk 2 above)
Comment (AlexanderMann) [reply tests/test_sandbox.py]: See our Slack convo. This just iterates over a real world stream which was snagged for testing etc.

I think this testing file should be reserved for cleansed real world data streams which we are only aiming to replay. The other testing files I think should have generators etc. which try to exercise the other various weirdness.

**Comment by awm33** [comment]
@AlexanderMann Just need to address tests

**Pull request merged**